command/pr_test.go

```diff
@@ -938,5 +939,125 @@ func TestPRReopen_alreadyMerged(t *testing.T) {
 	if !r.MatchString(err.Error()) {
 		t.Fatalf("output did not match regexp /%s/\n> output\n%q\n", r, output.Stderr())
 	}
+}
+
+type stubResponse struct {
+	ResponseCode int
+	ResponseBody io.Reader
+}
+
+func initWithStubs(branch string, stubs ...stubResponse) {
+	initBlankContext("", "OWNER/REPO", branch)
+	http := initFakeHTTP()
+	http.StubRepoResponse("OWNER", "REPO")
+
+	for _, s := range stubs {
+		http.StubResponse(s.ResponseCode, s.ResponseBody)
+	}
+}
+
+func TestPrMerge(t *testing.T) {
+	initWithStubs("master",
+		stubResponse{200, bytes.NewBufferString(`{ "data": { "repository": {
+			"pullRequest": { "number": 1, "closed": false, "state": "OPEN"}
+		} } }`)},
+		stubResponse{200, bytes.NewBufferString(`{"id": "THE-ID"}`)},
+	)
```

Comment:
I'm wondering what people think about this. Creating the blank context and the fake http instance seemed like it was getting very repetitive in tests with not much added benefit. Are helpers like this helpful?

Reply:
Indeed, it's getting really repetitive, especially because we are forced to copy-paste JSON response stubs for API calls that are not even being exercised in the current test. I think that `initWithStubs`, but not that we might be moving away from sequential stubs to those that match specific requests based on their contents https://github.com/cli/cli/pull/874 which means we might need access to additional apis other than just `http.StubResponse`